sale_order_line_price_history/wizards/sale_orderLine_price_history.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrderLinePriceHistory(models.TransientModel):
+    _name = "sale.order.line.price.history"
+    _description = "Sale order line price history"
+
+    @api.model
+    def _default_partner_id(self):
+        line_id = self.env.context.get("active_id")
+        return self.env['sale.order.line'].browse(line_id).order_partner_id.id
+
+    sale_order_line_id = fields.Many2one(
+        comodel_name='sale.order.line',
+        string='Sale order line',
+        default=lambda self: self.env.context.get("active_id"),
+    )
+    product_id = fields.Many2one(
+        related="sale_order_line_id.product_id",
+        readonly=True,
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Customer',
+        default=_default_partner_id,
+    )
+    line_ids = fields.One2many(
+        comodel_name="sale.order.line.price.history.line",
+        inverse_name="history_id",
+        string="History line",
+        readonly=True,
+    )
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        self.line_ids = False
+        domain = [
+            ("product_id", "=", self.product_id.id),
+            ("state", "in", ["sale", "done"]),
+        ]
+        if self.partner_id:
+            domain += [
+                "|",
+                ("order_id.partner_id", "=", self.partner_id.id),
```

Comment:
I have just seen that you have to put directly (and only this that includes all):
```python
("order_id.partner_id", "child_of", self.partner_id.commercial_partner_id.ids),
```